### ISSUE TITLE
Build failed for darwin at v3.6.4

### ISSUE THREAD

**Issue by suzuki-shunsuke** [sigi-cli/sigi#25]
Pre-built binaries for darwin weren't released.

https://github.com/sigi-cli/sigi/releases/tag/v3.6.4

<img width="968" alt="image" src="https://github.com/sigi-cli/sigi/assets/13323303/f1efbf35-31de-4fff-9401-38843d3ef88c">

The release workflow failed.

https://github.com/sigi-cli/sigi/actions/runs/9039178543/job/24841578216

```
error: linking with `/opt/osxcross/target/bin/x86_64-apple-darwin14-clang` failed: exit status: 1
  |
  = note: env -u IPHONEOS_DEPLOYMENT_TARGET -u TVOS_DEPLOYMENT_TARGET LC_ALL="C" PATH="/usr/local/rustup/toolchains/1.78.0-x86_64-unknown-linux-musl/lib/rustlib/x86_64-unknown-linux-musl/bin:/opt/osxcross/target/bin:/usr/local/cargo/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin" VSLANG="1033" ZERO_AR_DATE="1" "/opt/osxcross/target/bin/x86_64-apple-darwin14-clang" "-arch" "x86_64" "-m64" "/tmp/rustcjZKlbh/symbols.o" "/github/workspace/./target/x86_64-apple-darwin/release/deps/sigi-df1a5e23c45794bf.sigi.e4b0685584773015-cgu.0.rcgu.o" "-L" "/github/workspace/./target/x86_64-apple-darwin/release/deps" "-L" "/github/workspace/./target/release/deps" "-L" "/usr/local/rustup/toolchains/1.78.0-x86_64-unknown-linux-musl/lib/rustlib/x86_64-apple-darwin/lib" "/usr/local/rustup/toolchains/1.78.0-x86_64-unknown-linux-musl/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-d6175b4ffbd6f2cb.rlib" "-framework" "CoreFoundation" "-liconv" "-lSystem" "-lc" "-lm" "-L" "/usr/local/rustup/toolchains/1.78.0-x86_64-unknown-linux-musl/lib/rustlib/x86_64-apple-darwin/lib" "-o" "/github/workspace/./target/x86_64-apple-darwin/release/deps/sigi-df1a5e23c45794bf" "-Wl,-dead_strip" "-nodefaultlibs"
  = note: ld: warning: object file (/tmp/rustcjZKlbh/symbols.o) was built for newer macOS version (10.12) than being linked (10.8)
          ld: warning: object file (/github/workspace/./target/x86_64-apple-darwin/release/deps/sigi-df1a5e23c45794bf.sigi.e4b0685584773015-cgu.0.rcgu.o) was built for newer macOS version (10.12) than being linked (10.8)
          ld: warning: object file (/usr/local/rustup/toolchains/1.78.0-x86_64-unknown-linux-musl/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-d6175b4ffbd6f2cb.rlib(compiler_builtins-d6175b4ffbd6f2cb.compiler_builtins.c536cea08157bc6d-cgu.076.rcgu.o)) was built for newer macOS version (10.12) than being linked (10.8)
          Undefined symbols for architecture x86_64:
            "_clock_gettime", referenced from:
                std::time::SystemTime::now::h9a7358deb21206bf in sigi-df1a5e23c45794bf.sigi.e4b0685584773015-cgu.0.rcgu.o
            "_getentropy", referenced from:
                std::sys::thread_local::fast_local::Key$LT$T$GT$::try_initialize::h89972679ca8c9ab4 in sigi-df1a5e23c45794bf.sigi.e4b0685584773015-cgu.0.rcgu.o
          ld: symbol(s) not found for architecture x86_64
          clang-15: error: linker command failed with exit code 1 (use -v to see invocation)
          

error: could not compile `sigi` (bin "sigi") due to 1 previous error
Error:  Command exited with non-zero exit code
Error: Build failed
```

**Comment by booniepepper** [comment]
Thanks for the report @suzuki-shunsuke!

Looking into it, and it looks like I may need to look into using a mac runner instead of cross-compiling from Ubuntu

**Comment by booniepepper** [comment]
Updated in

https://github.com/sigi-cli/sigi/commit/b003d11f27ec2eef325ff03874d932aaa9c1ed88

And re-released v3.6.4, it now has the darwin binaries again. (And now supports AArch64 for macos too)
